source/particle/interpolator/quadratic_least_squares.cc

```diff
@@ -155,39 +250,398 @@ namespace aspect
+                        point[2] = -0.5;
+                        value_at_corner = evaluate(c[property_index], point);
+                        interpolation_min = std::min(interpolation_min, value_at_corner);
+                        interpolation_max = std::max(interpolation_max, value_at_corner);
+                        point[2] = 0.5;
+                        value_at_corner = evaluate(c[property_index], point);
+                        interpolation_min = std::min(interpolation_min, value_at_corner);
+                        interpolation_max = std::max(interpolation_max, value_at_corner);
+                        point[0] =  0.5;
+                        point[1] = -0.5;
+                        point[2] = -0.5;
+                        value_at_corner = evaluate(c[property_index], point);
+                        interpolation_min = std::min(interpolation_min, value_at_corner);
+                        interpolation_max = std::max(interpolation_max, value_at_corner);
+                        point[2] = 0.5;
+                        value_at_corner = evaluate(c[property_index], point);
+                        interpolation_min = std::min(interpolation_min, value_at_corner);
+                        interpolation_max = std::max(interpolation_max, value_at_corner);
+                        point[1] = 0.5;
+                        point[2] = -0.5;
+                        value_at_corner = evaluate(c[property_index], point);
+                        interpolation_min = std::min(interpolation_min, value_at_corner);
+                        interpolation_max = std::max(interpolation_max, value_at_corner);
+                        point[2] = 0.5;
+                        value_at_corner = evaluate(c[property_index], point);
+                        interpolation_min = std::min(interpolation_min, value_at_corner);
+                        interpolation_max = std::max(interpolation_max, value_at_corner);
+
+                      }
+                    if ((interpolation_max - cell_average_values[property_index]) > std::numeric_limits<double>::epsilon() &&
+                        (cell_average_values[property_index] - interpolation_min) > std::numeric_limits<double>::epsilon())
+                      {
+                        const double alpha = std::max(std::min((cell_average_values[property_index] - property_minimums[property_index])/(cell_average_values[property_index] - interpolation_min),
+                                                               (property_maximums[property_index]-cell_average_values[property_index])/(interpolation_max - cell_average_values[property_index])), 0.0);
+                        // If alpha >= 1, then using it would make the function grow from the cell average to meet the bounds.
+                        if (alpha < 1.0)
+                          {
+                            c[property_index] *= alpha;
+                            c[property_index][0] += (1-alpha) * cell_average_values[property_index];
```

Comment:
I understand the first line, but why do you have to modify the cell average (coefficient[0]) at all? Should'nt the average be conserved?

Reply:
Unlike bilinear least squares, in quadratic least squares c[0] cannot be expected to approach the cell average as the number of particles in a cell increases (keep this thought in mind as one of the changes @egpuckett and I are looking to make to bilinear will involve this idea). This same change applied later in the code could instead read 
```
interpolated_value = evalutate_...
interpolated_value = alpha * interpolated_value + (1-alpha) * cell_average
```
As alpha decreases, the output of the limiter scheme approaches cell average.